Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py

```diff
@@ -1955,47 +1959,75 @@ def update_remote_system_command(client, args):
-        else:
-            incident_data = raw_incident
+    try:
+        for raw_incident in raw_incidents:
+            incident_id = raw_incident.get('incident_id')
 
-        sort_all_list_incident_fields(incident_data)
+            if demisto.params().get('extra_data'):
+                incident_data = get_incident_extra_data_command(client, {"incident_id": incident_id,
+                                                                         "alerts_limit": 1000})[2].get('incident')
+            else:
+                incident_data = raw_incident
 
-        incident_data['mirror_direction'] = MIRROR_DIRECTION.get(demisto.params().get('mirror_direction', 'None'), None)
-        incident_data['mirror_instance'] = demisto.integrationInstance()
+            sort_all_list_incident_fields(incident_data)
 
-        description = raw_incident.get('description')
-        occurred = timestamp_to_datestring(raw_incident['creation_time'], TIME_FORMAT + 'Z')
-        incident = {
-            'name': f'#{incident_id} - {description}',
-            'occurred': occurred,
-            'rawJSON': json.dumps(incident_data),
-        }
+            incident_data['mirror_direction'] = MIRROR_DIRECTION.get(demisto.params().get('mirror_direction', 'None'),
+                                                                     None)
+            incident_data['mirror_instance'] = demisto.integrationInstance()
 
-        if demisto.params().get('sync_owners') and incident_data.get('assigned_user_mail'):
-            incident['owner'] = demisto.findUser(email=incident_data.get('assigned_user_mail')).get('username')
+            description = raw_incident.get('description')
+            occurred = timestamp_to_datestring(raw_incident['creation_time'], TIME_FORMAT + 'Z')
+            incident = {
+                'name': f'#{incident_id} - {description}',
+                'occurred': occurred,
+                'rawJSON': json.dumps(incident_data),
+            }
 
-        # Update last run and add incident if the incident is newer than last fetch
-        if raw_incident['creation_time'] > last_fetch:
-            last_fetch = raw_incident['creation_time']
+            if demisto.params().get('sync_owners') and incident_data.get('assigned_user_mail'):
+                incident['owner'] = demisto.findUser(email=incident_data.get('assigned_user_mail')).get('username')
+
+            # Update last run and add incident if the incident is newer than last fetch
+            if raw_incident['creation_time'] > last_fetch:
+                last_fetch = raw_incident['creation_time']
+
+            incidents.append(incident)
+            non_created_incidents.remove(raw_incident)
+
+    except Exception as e:
+        if "Rate limit exceeded" in str(e):
+            # To prevent an infinite loop without erroring out
+            if non_created_incidents == incidents_from_previous_run:
```

Comment:
there is a chance that the next run will also get Rate limit exceeded for every incident.
remove this condition.
writing to the logs to should be enough